java/lang/file_permission_others.yml

```
@@ -0,0 +1,43 @@
+patterns:
+  - pattern: |
+      {
+        $<!>$<PERMISSIONS:identifier>.add($<OTHER_PERMISSION>);
+        Files.setPosixFilePermissions($<_>, $<PERMISSIONS>);
+      }
+    filters:
+      - variable: PERMISSIONS
+        detection: java_lang_get_posix_file_permissions
+      - variable: OTHER_PERMISSION
+        regex: ^(PosixFilePermission.)?OTHER_
+auxiliary:
+  - id: java_lang_get_posix_file_permissions
+    patterns:
+      - pattern: $<FILES>.getPosixFilePermissions();
+        filters:
+          - variable: FILES
+            regex: ^(java.)?(nio.)?(file.)?Files
+languages:
+  - java
+severity: warning
+metadata:
+  description: "File permission open to other detected."
```

It might read a little better with quotes, or uppercase:
```suggestion
  description: "File permission open to 'other' detected."
```